src/app/app.component.html

```diff
@@ -0,0 +1,54 @@
+<div class="mat-app-background">
+  <div class="header">
+    <mat-toolbar class="toolbar" color="primary">
+      <h1 class="toolbar__title">{{ title }}</h1>
+      <div class="toolbar__options">
+        <app-filter (filterChange)="onFilterChange($event)"></app-filter>
+      </div>
+    </mat-toolbar>
+    <mat-progress-bar *ngIf="loading" class="progress-bar" mode="indeterminate"></mat-progress-bar>
+  </div>
+
+  <table class="table" mat-table matSort [dataSource]="dataSource">
+    <ng-container *ngFor="let column of displayedColumns" [matColumnDef]="column">
+      <th
+        mat-header-cell
+        *matHeaderCellDef
+        mat-sort-header
+        [sortActionDescription]="'Sort by ' + column"
+      >
+        {{ column.toUpperCase() }}
+      </th>
+      <div *ngIf="column !== 'type'">
```

Comment:
`type` is a magic string, replace it with a constant variable like
`const TYPE = 'type';`
use TYPE

or create an ENUM if you have to do this with multiple strings